embassy-usb/src/class/hid.rs

```diff
@@ -106,13 +137,15 @@ fn build<'d, D: Driver<'d>>(
     state: &'d mut State<'d>,
     config: Config<'d>,
     with_out_endpoint: bool,
+    usb_subclass: u8,
+    usb_protocol: u8,
 ) -> (Option<D::EndpointOut>, D::EndpointIn, &'d AtomicUsize) {
     let len = config.report_descriptor.len();
 
-    let mut func = builder.function(USB_CLASS_HID, USB_SUBCLASS_NONE, USB_PROTOCOL_NONE);
+    let mut func = builder.function(USB_CLASS_HID, usb_subclass, usb_protocol);
```

Comment:
Is there any particular reason that `config.device_sub_class` and `config.device_protocol` are not used here?